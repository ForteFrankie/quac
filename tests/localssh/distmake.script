#!/bin/bash

# Test that distributed make over ssh works.

# Copyright (c) 2012-2013 Los Alamos National Security, LLC, and others.

. ./environment.sh

cd $DATADIR
exec 2>&1

# Can we ssh to localhost without a password? That's a prerequisite for all
# the fancy stuff below...
y "sshrot echo a"

# Should be using SSH (not MPI)
x sshrot --info

# Same working directory test. Note that Python's os.getcwd() can give
# different results than pwd if symlinks are involved (though we use pwd under
# sshrot as it's already passed through os.getcwd() at that point).
y "python -c 'import os; print(os.getcwd())' > pwd_local"
y 'sshrot pwd > pwd_sshrot'
y 'diff -u pwd_local pwd_sshrot'

# QUACreduce over ssh test
echo -e 'foo bar baz\nfoo foo' > foo1.txt
echo -e 'bar' > foo2.txt
quacreduce --dist \
           --map 'tr "[:blank:]" "\n"' \
           --reduce 'uniq -c > out/%(RID)' \
           --partitions 3 \
           foo*.txt
x make -j3 --quiet
y 'cat out/* | sort'
x sshrot --cleanup
