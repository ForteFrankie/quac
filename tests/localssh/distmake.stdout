$ (sshrot echo a)
a
$ sshrot --info
SSH
$ sshrot --muxstart
$ (sshrot 'echo b && sleep 2 && echo c' &)
$ (sshrot 'echo b && sleep 3 && echo d' &)
$ sleep 1.5
b
b
ssh clients:      2
ssh masters:      1
control sockets:  1
TCP connections:  1
$ sleep 1.0
c
ssh clients:      1
ssh masters:      1
control sockets:  1
TCP connections:  1
$ sleep 1.5
d
$ sshrot --muxstop
Exit request sent.
ssh clients:      0
ssh masters:      0
control sockets:  0
TCP connections:  0
$ sshrot --muxstart
$ (sshrot 'sleep 1 && echo e' &)
$ (sshrot 'sleep 1 && echo e' &)
$ (sshrot 'sleep 1 && echo e' &)
$ (sshrot 'sleep 1 && echo e' &)
$ (sshrot 'sleep 1 && echo e' &)
$ (sshrot 'sleep 1 && echo e' &)
$ (sshrot 'sleep 1 && echo e' &)
$ (sshrot 'sleep 1 && echo e' &)
$ (sshrot 'sleep 1 && echo e' &)
$ sleep 0.5
ssh clients:      9
ssh masters:      1
control sockets:  1
TCP connections:  1
$ sleep 1.0
e
e
e
e
e
e
e
e
e
$ sshrot --muxstop
Exit request sent.
$ (python -c 'import os; print os.getcwd()' > pwd_local)
$ (sshrot pwd > pwd_sshrot)
$ (diff -u pwd_local pwd_sshrot)
$ sshrot --muxstart
$ make -j3 --quiet
$ (cat out/* | sort)
      1 baz
      2 bar
      3 foo
$ sshrot --muxstop
Exit request sent.
